lms/djangoapps/discussion_api/tests/test_api.py

```diff
@@ -448,6 +462,23 @@ def test_thread_content(self):
             }
         )
 
+    @ddt.data(
+        (FORUM_ROLE_ADMINISTRATOR, True, False),
+        (FORUM_ROLE_MODERATOR, True, False),
+        (FORUM_ROLE_COMMUNITY_TA, True, False),
+        (FORUM_ROLE_STUDENT, True, True),
+        (FORUM_ROLE_STUDENT, False, False),
+    )
```

Comment:
It initially struck me as unnecessary since it seems really unlikely that there would be a bug that would cause a group_id to not be sent for a student but be sent for a privileged user, but I think you are right that it's better to be comprehensive.

As for using `itertools.product`, the downside there is that `expected_has_group` then needs to be computed from the other fields rather than passed as an argument. On the one hand, I do like the compactness of using `product`, but on the other hand I really like having the expectation declared alongside the rest of the variables in the test case. My main concern with computing the expectation is that it repeats the logic of the code under test, and there's some risk of having matching bugs in both places.